### ISSUE TITLE
Add "masked" to the loss function names

### ISSUE THREAD

**Issue by pplantinga** [speechbrain/speechbrain#208]
Changing the names of the loss functions could be nice for helping users understand how our loss functions differ from the pytorch loss functions. The new names could be something like:

* `masked_l1`
* `masked_mse`
* `masked_error`
* `masked_nll`
